source/beamline/BioXAS/BioXASSideMonochromator.cpp

```diff
@@ -78,124 +78,178 @@ BioXASSideMonochromator::~BioXASSideMonochromator()
 
 }
 
+AMAction3* BioXASSideMonochromator::createSetEnergyAction(double newEnergy)
+{
+	if (!energy_->isConnected())
+		return 0;
+
+	return AMActionSupport::buildControlMoveAction(energy_, newEnergy);
+}
+
 AMAction3* BioXASSideMonochromator::createCloseSlitsAction()
 {
-	if (!slitsClosed_->isConnected())
+	if (!slitsStatus_->isConnected())
 		return 0;
 
-	return AMActionSupport::buildControlMoveAction(slitsClosed_, 1);
+	return AMActionSupport::buildControlMoveAction(slitsStatus_, Slits::Closed);
 }
 
 AMAction3* BioXASSideMonochromator::createRemovePaddleAction()
 {
-	if (!paddleOut_->isConnected())
+	if (!paddleStatus_->isConnected())
 		return 0;
 
-	return AMActionSupport::buildControlMoveAction(paddleOut_, 1);
+	return AMActionSupport::buildControlMoveAction(paddleStatus_, Paddle::Out);
 }
 
-AMAction3* BioXASSideMonochromator::createWaitForCrystalChangeEnabledAction()
+AMAction3* BioXASSideMonochromator::createMoveBraggMotorAction(double degDestination)
 {
-	if (!crystalChangeEnabled_->isConnected())
+	if (!braggAbsolutePosition_->isConnected())
 		return 0;
 
-	AMControlInfo setpoint = crystalChangeEnabled_->toInfo();
-	setpoint.setValue(1);
-	AMControlWaitActionInfo *actionInfo = new AMControlWaitActionInfo(setpoint);
-	AMControlWaitAction *action = new AMControlWaitAction(actionInfo, crystalChangeEnabled_);
+	return AMActionSupport::buildControlMoveAction(braggAbsolutePosition_, degDestination);
+}
 
-	return action;
+AMAction3* BioXASSideMonochromator::createMoveCrystalChangeMotorAction(int relDestination)
+{
+	if (!crystalChangeMotorRel_->isConnected())
+		return 0;
+
+	return AMActionSupport::buildControlMoveAction(crystalChangeMotorRel_, relDestination, true);
 }
 
-AMAction3* BioXASSideMonochromator::createMoveToCrystalChangePositionAction()
+AMAction3* BioXASSideMonochromator::createSetBraggMotorPowerOnAction()
 {
-	return createMoveStageAction(55);
+	return AMActionSupport::buildControlMoveAction(braggMotorPower_, 1);
```

Comment:
Just a question here: is it possible to make an enum definition for braggMotorPower status (the magic number "1" and "3" below) as what you did for the others (sucha as paddle, slit above)?